solvation_analysis/solvation.py

```diff
@@ -125,7 +134,18 @@ def get_closest_n_mol(u, central_species, n_mol=5, radius=3):
     ]
     str_resids = " ".join(str(resid) for resid in ordered_resids[closest_n_resix])
     full_shell = u.select_atoms(f"resid {str_resids}")
-    return full_shell, ordered_resids[closest_n_resix], radii[ordering][closest_n_resix]
+    if return_resids and return_radii:
```

Comment:
It looks to me as if we are sorting only by resid right? What if I just wanted the closest n atoms regardless of their resid? I think we should try and make this function very general as its your main workhorse. Other option is perhaps change its name to `get_closest_n_res` or something like that.

Perhaps taking a kwarg on which to sort could be a solution like:

```python
def get_closest_n_mol(u, central_species, n_mol=5, radius=3, sort="resids", return_resids=False, return_radii=False):
``` 

 I would also be interested what @richardjgowers and/ or @IAlibay think.

Reply:
Maybe @orioncohen wants to provide more information on what the intended use of this function is.

For the moment it doesn't seem very generalizable (particularly given that it just picks up any molecule rather than a defined type) but maybe I'm missing the point of how it fits with other things.

Reply:
Originally I envisioned this function to select the closest n molecules, which I used resids to achieve. My thought was to use this to provide a similar functionality to get_radial_shell with more consistent return values. If I want to build an analysis tool that spans multiple time steps, it's helpful to have a set return length. I've often used this function to select a specific number of solvating molecules for visualizations. I definitely agree that it could be generalized and I like @hmacdope's suggestion to include a kwarg.

As a molecular chemist, I only need to think in terms of atoms and molecules and I don't use the protein and segment features of MDA. That's a bit out of step with the MDA community at large, so I'd appreciate feedback on how I can make this more flexible and more broadly usable.

Reply:
I guess seeing as we are specialising in solvation here, atoms and molecules is fine. Perhaps a more general case is a possible enhancement to add down the track.

Reply:
Sorry by "a defined type" I meant that usually with a function like this you'd often want to do "find the nearest N waters to ion" rather than "find all molecules of any type nearest to ion". My understanding is that this function does the latter rather than the former.

Personally I'd switch this to using an atomgroup input argument and then doing a loop that calls `capped_distance`, and calculates the number of "things" (make it optional so that you can either pick up atoms, residues, whole molecules, etc...) from the capped_distance return [I guess the point I'm making is that residues != molecules in many cases].

Since this PR is more about getting pytest going - I'd say let's just revisit this in a specific PR / when you switch over to capped_distance.

Reply:
Agreed lets revisit in a new PR once we have final tests.

Reply:
Quick question here. Is there a canonical way of adding optional return values? The set of conditionals that I am using here feels really sloppy.

As @hmacdope  suggested, I'll look into generalizing this function in another PR.

Reply:
I'm going to mark this as resolved in that case.